src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/HashPartitioningConditionMapper.java

```diff
@@ -62,15 +64,23 @@
         return fieldPartners;
     }
 
+    protected void validateFieldsCanBeUpdated(Set<String> allUpdateFields) {
+        validateFieldsCanBeUpdated(allUpdateFields, UpdateType.SET);
+    }
+
     @Override
-    protected void validateFieldsCanBeUpdated(Set<String> allSetFields) {
-        super.validateFieldsCanBeUpdated(allSetFields);
+    protected void validateFieldsCanBeUpdated(Set<String> allUpdateFields, UpdateType type) {
+        super.validateFieldsCanBeUpdated(allUpdateFields, type);
 
-        for (String field : allSetFields) {
+        for (String field : allUpdateFields) {
+            if (type.equals(UpdateType.ADD)) {
+                secondaryIndexes.stream().forEach(index -> validateUpdatedFieldIsNotInIndexKey(
```

Comment:
It will fail the validation if the index field is part of the hash or range key for the index.  I think what you're referring to is the primary key validation (unrelated to indexes?), which is checked in this method and called on super in this case `validateFieldsCanBeUpdated`
https://github.com/salesforce/mt-dynamo/blob/351beab4d6522b94d4a56e1ba7820d871b1dabfa/src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/AbstractConditionMapper.java#L135

or are you referring to something else?